DOCS-#3795: Expand `using_pandas_on_dask.rst` doc (#3808)
modin-project/modin: docs/developer/using_pandas_on_dask.rst

pandas on Dask
==============

This section describes usage related documents for the pandas on Dask component of Modin.

Modin uses pandas as a primary memory format of the underlying partitions and optimizes queries
ingested from the API layer in a specific way to this format. Thus, there is no need to care of choosing it
but you can explicitly specify it anyway as shown below.

> One of the execution engines that Modin uses is Dask. To enable this engine you should set the following environment variables:

Reviewer: ```suggestion
One of the execution engines that Modin uses is Dask. To enable the pandas on Dask execution you should set the following environment variables:
```

Author: applied
